### PR TITLE
Remove permissions from tar file

### PR DESCRIPTION
**Pull request by dfeldman**
Adding --owner=root and --group=root to the tar file means that the tar file is in preserve-permissions mode. That means that if you extract it in your home directory, it will change the permissions of your home directory, and potentially prevent login!* I think this was needed because we used to extract the tar file to create the docker images, but now they are created separately and the tar files are not used by anything other than end users.

   * (If your home directory is 755, sshd will stop you from logging in. So you could get locked out of your whole account)

Signed-off-by: Daniel Feldman <dfeldman.mn@gmail.com>

**Pull Request check list**

- [ ] Commit conforms to CONTRIBUTING.md?
- [ ] Proper tests/regressions included?
- [ ] Documentation updated?

**Affected functionality**

**Description of change**

**Which issue this PR fixes**

### CODE REVIEW

**Comment by azdagron** [comment]
Yuck. That's unfortunate! Fortunately this should only happen if you extract as root (e.g., with `sudo`).

This is a side effect of how the tarball is created. It makes an entry for `.`, which is what leads to the ownership change.

I don't know if the owner and group setting is really required anymore, but if we're just interested in fixing the bug on extraction, we can change the way the tarballs are created to not include `.`. IMO, we'd want to not include `.` even if we changed the user and group settings.

Something like:

```$ git diff
diff --git a/script/build-artifact.sh b/script/build-artifact.sh
index 4571ed70..d1b86c60 100755
--- a/script/build-artifact.sh
+++ b/script/build-artifact.sh
@@ -56,7 +56,7 @@ cp "${BINDIR}"/oidc-discovery-provider "${EXTRAS_STAGING}"/bin
 
 # Create the tarballs and checksums
 mkdir -p "${OUTDIR}"
-tar -cvzf "${TARBALL}" --directory "${TMPDIR}/spire" "${TAROPTS[@]}" .
+(cd "${TMPDIR}/spire"; tar -cvzf "${TARBALL}" "${TAROPTS[@]}" *)
 echo "$(shasum -a 256 "${TARBALL}" | cut -d' ' -f1) $(basename "${TARBALL}")" > "${CHECKSUM}"
-tar -cvzf "${EXTRAS_TARBALL}" --directory "${TMPDIR}/spire-extras" "${TAROPTS[@]}" .
+(cd "${TMPDIR}/spire-extras"; tar -cvzf "${EXTRAS_TARBALL}" "${TAROPTS[@]}" *)
 echo "$(shasum -a 256 "${EXTRAS_TARBALL}" | cut -d' ' -f1) $(basename "${EXTRAS_TARBALL}")" > "${EXTRAS_CHECKSUM}"
```

**Comment by dfeldman** [comment]
Yeah, removing . would definitely work too. Which do you think is better?

**Comment by azdagron** [comment]
Sorry, yes, root access (e.g. `sudo`)  would be required for `tar` to change ownership to root, but yeah, of course tar is able to change mode bits for files and directories the invoker has permissions to change.

I think we need to remove `.` either way.

As far as changing --owner and --group, I think the original intent there was to not have the username of whoever built the release to be preserved in the tarball. It isn't that big of a deal really if we remove it, but I don't think it hurts anything as long as we fix `.`?

**Comment by dfeldman** [comment]
Got it! I will change it to remove "." from the tar file & do some experiments to make certain it works.

**Comment by azdagron** [comment]
Great! Thanks, Dan.

**Comment by azdagron** [comment]
In order to get Travis to run, you'll need to merge in, or rebase on top of, the latest `main` branch. We renamed `master` to `main` recently and the requisite changes to update scripts and what not were just merged.

**Comment by azdagron** [comment]
I think something is awry with the way you've pulled in the changes from `main`. Since this change is so tiny, maybe the easiest thing to do is to just rebase your change on `main` and force push over your PR branch?

**Comment by dfeldman** [comment]
Thanks, yeah, I did it wrong. I force pushed, I think this should work.

File: script/build-artifact.sh [hunk 1]
```diff
@@ -18,7 +18,6 @@ if [ "$OS" = "linux" ]; then
         *muslr*) LIBC="-musl" ;;
         *) LIBC="-unknown" ;;
     esac
-    TAROPTS=("--owner=root" "--group=root")
```
Comment (azdagron) [inline script/build-artifact.sh:21]: Sorry, I think we had a miscommunication. I don't think we want to change the ownership. We don't want different user/group settings to leak into the tarball dependent on the build environment.

**Comment by azdagron** [comment]
Shellcheck linting is failing on the use of `*`

**Comment by dfeldman** [comment]
(Hopefully) should be good to go

File: script/build-artifact.sh [hunk 2]
```diff
@@ -56,7 +56,7 @@ cp "${BINDIR}"/oidc-discovery-provider "${EXTRAS_STAGING}"/bin
 
 # Create the tarballs and checksums
 mkdir -p "${OUTDIR}"
-tar -cvzf "${TARBALL}" --directory "${TMPDIR}/spire" "${TAROPTS[@]}" .
+(cd "${TMPDIR}/spire"; tar -cvzf "${TARBALL}" "${TAROPTS[@]}" ./*)
```
Comment (azdagron) [inline script/build-artifact.sh:59]: nitpicky, but can we do the other option suggested by shellcheck? The current method still leads to each entry being appended with `./`, which isn't a huge deal, but I'd rather just steer clear:
```suggestion
(cd "${TMPDIR}/spire"; tar -cvzf "${TARBALL}" "${TAROPTS[@]}" -- *)
```

File: script/build-artifact.sh [hunk 3]
```diff
@@ -56,7 +56,7 @@ cp "${BINDIR}"/oidc-discovery-provider "${EXTRAS_STAGING}"/bin
...
 echo "$(shasum -a 256 "${TARBALL}" | cut -d' ' -f1) $(basename "${TARBALL}")" > "${CHECKSUM}"
-tar -cvzf "${EXTRAS_TARBALL}" --directory "${TMPDIR}/spire-extras" "${TAROPTS[@]}" .
+(cd "${TMPDIR}/spire-extras"; tar -cvzf "${EXTRAS_TARBALL}" "${TAROPTS[@]}" ./*)
```
Comment (azdagron) [inline script/build-artifact.sh:61]:
```suggestion
(cd "${TMPDIR}/spire-extras"; tar -cvzf "${EXTRAS_TARBALL}" "${TAROPTS[@]}" -- *)
```

**Review by azdagron — APPROVED** [review]
Thanks, @dfeldman ! Appreciate your patience with the back-and-forth.

**Pull request merged**